plugins/out_kinesis_firehose/firehose_api.c

```diff
@@ -235,11 +248,10 @@ static int process_event(struct flb_firehose *ctx, struct flush *buf,
         time_key_ptr += 3;
         tmp_size = buf->tmp_buf_size - buf->tmp_buf_offset;
         tmp_size -= (time_key_ptr - tmp_buf_ptr);
-        len = strftime(time_key_ptr, tmp_size, ctx->time_key_format, &time_stamp);
-        if (len <= 0) {
-            /* ran out of space - should not happen because of check above */
-            return 1;
-        }
+
+        /* merge out_buf to time_key_ptr */
+        strncpy(time_key_ptr, out_buf, len);
```

Comment:
The records prepped for the request are not null terminated IIRC/from quickly re-reading the surrounding code. I thought this would cause a problem, but then I read that strncpy will not add any terminating null byte if the source string is `len` chars long without the null byte. 

So this means this call is exactly equivalent to:

```
memcpy(time_key_ptr, out_buf, len);
```